snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/SnomedEditingContext.java

```diff
@@ -619,126 +628,96 @@ public SnomedDeletionPlan canDelete(Concept concept, SnomedDeletionPlan deletion
-
+		
 		deletionPlan.markForDeletion(referringMembers);
 		deletionPlan.markForDeletion(concept);
-		return deletionPlan;
 	}
 
-	public SnomedDeletionPlan canDelete(Description description, SnomedDeletionPlan deletionPlan, boolean force) {
-		// If the description is the target of the deletion, check validity
-		if (deletionPlan == null) {
-			deletionPlan = new SnomedDeletionPlan();
-			
-			// Check if description is already released, and this is not a forced delete
-			if (description.isReleased() && !force) {
-				deletionPlan.addRejectionReason(String.format(COMPONENT_IS_RELEASED_MESSAGE, "description", toString(description)));
-				return deletionPlan;
-			}
+	private void canDelete(Description description, boolean force) {
 			
-			// not the only fully specified name
-			if (FULLY_SPECIFIED_NAME.equals(description.getType().getId())) {
-				boolean hasOtherFullySpecifiedName = false;
-				final List<Description> otherDescriptions = description.getConcept().getDescriptions();
-				for (Description otherDescription: otherDescriptions) {
-					// another fully specified name exists that is not this description
-					if (FULLY_SPECIFIED_NAME.equals(otherDescription.getType().getId()) && description != otherDescription) {
-						hasOtherFullySpecifiedName = true;
-						break;
-					}
-				}
-				
-				if (!hasOtherFullySpecifiedName) {
-					deletionPlan.addRejectionReason(UNABLE_TO_DELETE_ONLY_FSN_DESCRIPTION_MESSAGE);
-					return deletionPlan;
-				}
-			}
-		} 
+		// Check if description is already released, and this is not a forced delete
+		if (description.isReleased() && !force) {
+			deletionPlan.addRejectionReason(String.format(COMPONENT_IS_RELEASED_MESSAGE, "description", toString(description)));
+			return;
+		}
 		
 		deletionPlan.markForDeletion(refSetEditingContext.getReferringMembers(description));
 		deletionPlan.markForDeletion(description);
-		return deletionPlan;
 	}
 
-	public SnomedDeletionPlan canDelete(Relationship relationship, SnomedDeletionPlan deletionPlan, boolean force) {
-		// If the relationship is the target of the deletion, check validity
-		if (deletionPlan == null) {
-			deletionPlan = new SnomedDeletionPlan();
+	private void canDelete(Relationship relationship, boolean force) {
 		
-			// Check if description is already released, and this is not a forced delete
-			if (relationship.isReleased() && !force) {
-				deletionPlan.addRejectionReason(String.format(COMPONENT_IS_RELEASED_MESSAGE, "relationship", toString(relationship)));
-				return deletionPlan;
-			}
+		// Check if description is already released, and this is not a forced delete
+		if (relationship.isReleased() && !force) {
+			deletionPlan.addRejectionReason(String.format(COMPONENT_IS_RELEASED_MESSAGE, "relationship", toString(relationship)));
+			return;
 		}
 		
-		deletionPlan.markForDeletion(refSetEditingContext.getReferringMembers(relationship));		
+		deletionPlan.markForDeletion(refSetEditingContext.getReferringMembers(relationship));
```

Comment:
Further performance improvements could be made here by moving the `getReferringMembers` logic to the `preCommit` phase just like we did it with the actual component deletion.